[WebGPU] Indirect indexed validation does not take into account index buffer offset

mwyrzykowski opened the pull request:
#### e3bf879afc70b1c753bc4f0bea2b19f7cf911ee0
<pre>
[WebGPU] Indirect indexed validation does not take into account index buffer offset
<a href="https://bugs.webkit.org/show_bug.cgi?id=281539">https://bugs.webkit.org/show_bug.cgi?id=281539</a>
<a href="https://rdar.apple.com/137966310">rdar://137966310</a>

Reviewed by Tadeu Zagallo.

drawIndirectIndexed must have space for at least one index

* LayoutTests/fast/webgpu/nocrash/fuzz-281539-expected.txt: Added.
* LayoutTests/fast/webgpu/nocrash/fuzz-281539.html: Added.
Add regression test.

* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::clampIndirectIndexBufferToValidValues):

Canonical link: <a href="https://commits.webkit.org/285260@main">https://commits.webkit.org/285260@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/62926bd65f538ed12045f67f15362af3b96a4f4e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/71961 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/51381 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/24746 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/76121 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/23170 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/59182 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/22990 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/56774 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/15281 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/75028 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/46597 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/61998 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/37207 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/43260 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/19468 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/21515 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/65167 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/19832 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/77801 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/16201 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/19001 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/16247 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/62021 "Passed tests") | [⏳ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/API-Tests-GTK-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15916 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/12701 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/6351 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/47179 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/48248 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/49535 "Built successfully") | | | 
| | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/47992 "Build is in progress. Recent messages:OS: Sonoma (14.5), Xcode: 15.4; Running apply-patch; Checked out pull request; Compiled WebKit (warnings)") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ae9bd31c01eed5f40f3a504ea68a2652f93542ab. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/35249

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/62926bd65f538ed12045f67f15362af3b96a4f4e. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/35249

mwyrzykowski committed:
[WebGPU] Indirect indexed validation does not take into account index buffer offset
https://bugs.webkit.org/show_bug.cgi?id=281539
rdar://137966310

Reviewed by Tadeu Zagallo.

drawIndirectIndexed must have space for at least one index

* LayoutTests/fast/webgpu/nocrash/fuzz-281539-expected.txt: Added.
* LayoutTests/fast/webgpu/nocrash/fuzz-281539.html: Added.
Add regression test.

* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::clampIndirectIndexBufferToValidValues):

Canonical link: https://commits.webkit.org/285260@main

webkit-commit-queue commented:
Committed 285260@main (e3bf879afc70): <https://commits.webkit.org/285260@main>

Reviewed commits have been landed. Closing PR #35249 and removing active labels.
